pkg/kapp/preflight/registry.go

```diff
@@ -101,9 +102,56 @@ func (c *Registry) AddCheck(name string, check Check) {
 	if c.known == nil {
 		c.known = make(map[string]Check)
 	}
+	if c.enabledFlag == nil {
+		c.enabledFlag = make(map[string]bool)
+	}
 	c.known[name] = check
 }
 
+// Validate the configuration provided; the rules are:
+// 1. Unknown validator = error
+// 2. Duplicate validator = error
+func (c *Registry) validateConfig(conf config.Conf) error {
+	haveConfig := map[string]bool{}
+	for _, rule := range conf.PreflightRules() {
+		if _, ok := c.known[rule.Name]; !ok {
+			return fmt.Errorf("unknown preflight check in configuration: %q", rule.Name)
+		}
+		if _, ok := haveConfig[rule.Name]; ok {
+			return fmt.Errorf("duplicate preflight check in configuration: %q", rule.Name)
+		}
+		haveConfig[rule.Name] = true
+	}
+	return nil
+}
+
+func (c *Registry) SetConfig(conf config.Conf) error {
```

Comment:
@tmshort I know we have already had this conversation in the PR against my fork before the CRD proposal was approved, but I wanted to call it out here and get the opinions of the carvel maintainers as well.

IMO we should change this to not accept the full `config.Conf` type. Instead I think something like:
```suggestion
func (c *Registry) SetRules(rules ...config.PreflightRule) error {
```
would be better here for two reasons:
1. This method really only cares about the preflight rules anyways. Because of this it is easier and more clear to a user what information is really used and taken into account when using this method to configure a preflight registry. Keeping the parameters to the minimum thing needed is a better UX IMO.
2. Eliminates the parsing of YAML in the unit tests, reducing the, IMO, unrelated points of failure in the unit tests. I feel that changes to logic that would break the YAML parsing should break tests specifically associated with parsing of the YAML contents and not the tests for a function that assumes the parsing has already been done and takes in the type resulting from the YAML parsing.

I'm by no means saying this can't go in like this, I'm totally fine with that if the consensus is that what is currently implemented is the best approach. Just wanted to highlight what I think might make things a bit easier to understand and reason about as both a consumer of the library and a maintainer.

Reply:
If this were the desired route, I'd prefer `[]PrelightRule`, as that's the output of `config.PreflightRules()`. The above approach requires an unnecessary break-apart step, vs:
```
registry.SetConfig(conf.PreflightRules())
```

Reply:
> I'd prefer []PrelightRule as the parameter, as that's the output of config.PreflightRules(). The above approach requires an unnecessary break-apart step, vs

Fair point, totally fine with me. I was just throwing something together as an example.

>I prefer the YAML config in the unit tests, as I think it's easier to read than golang structs.

Also fair, but I personally feel its easier to read the golang structs than the yaml in this case. To each their own :). This also isn't explicitly about removing the YAML config, I'm fine with that approach if the consensus is that the currently implemented approach. I personally think the removal of the YAML config is one benefit, but I think the other benefit I mentioned stands on its own.

Reply:
Personally, I prefer the YAML config in the tests as well, as it's easier to scan through them, however, I agree with @everettraven that we need not pass the whole config and just the preflight rules should be enough.

Reply:
I can tweak the API to only pass the preflight rules.